plugin.py

```diff
@@ -18,21 +21,29 @@ class LspPyrightPlugin(NpmClientHandler):
     server_directory = "language-server"
     server_binary_path = os.path.join(server_directory, "node_modules", "pyright", "langserver.index.js")
 
-    @classmethod
-    def additional_variables(cls) -> Optional[Dict[str, str]]:
-        variables = super().additional_variables() or {}
-        variables.update(
-            {
-                "sublime_py_files_dir": os.path.dirname(sublime.__file__),
-            }
-        )
-
-        return variables
-
     @classmethod
     def install_in_cache(cls) -> bool:
         return False
 
     @classmethod
     def minimum_node_version(cls) -> Tuple[int, int, int]:
         return (12, 0, 0)
+
+    @classmethod
+    def on_client_configuration_ready(cls, configuration: Dict[str, Any]) -> None:
+        super().on_client_configuration_ready(configuration)
+
+        if get_setting("st_plugin_development_mode"):
+            configuration["settings"]["python.analysis.extraPaths"].extend(cls.find_package_dependency_dirs())
```

Comment:
If the user's `LSP-pyright.sublime-setting` looks like this:

```js
{
    "settings":
    {
    },
}
```

then this code will crash because `configuration` available here is not yet merged with the base configuration. So you should be more careful and create the `settings` and `python.analysis.extraPaths` objects if necessary.

Reply:
Confirmed for settings in `on_settings_read()` and resolved in https://github.com/sublimelsp/LSP-pyright/pull/26/commits/1b7a8587c4936adaaadb0eb766e64c8add274769